src/mca/ptl/tcp/ptl_tcp.c

```diff
@@ -1315,21 +1318,6 @@ static pmix_status_t recv_connect_ack(int sd, uint8_t myflag)
                             pmix_globals.myid.nspace, pmix_globals.myid.rank,
                             pmix_client_globals.myserver->info->pname.nspace,
                             pmix_client_globals.myserver->info->pname.rank);
-
```

Comment:
IIUC we don't have psec modules with handshake so far.
So this change should be fine.

One clarification:
It seems to me that this code is not quite correct:
* Status message is received above (generic code for both client and server)
* but here you do another receive before and not after client's handshake.
If I understand the right way would be to perform the handshake and only afterward exchange with messages.

Reply:
The handshake itself exchanges messages as part of its procedure. I'd have to look at the broader code to understand the situation.

Reply:
So let me rephrase:
Does it make sense to send the message before server has completed handshake?
IMO it doesn't, but it seems to be here in this sequence.

Reply:
Bringing the whole piece in question here:
```C
        pmix_ptl_base_recv_blocking(sd, (char*)&reply, sizeof(pmix_status_t));	
        if (PMIX_SUCCESS != reply) {	
            /* see if they want us to do the handshake */	
            if (PMIX_ERR_READY_FOR_HANDSHAKE == reply) {	
                PMIX_PSEC_CLIENT_HANDSHAKE(reply, pmix_client_globals.myserver, sd);	
                if (PMIX_SUCCESS != reply) {	
                    return reply;	
                }	
                /* if the handshake succeeded, then fall thru to the next step */	
            } else {	
                return reply;	
            }	
        }
```

Reply:
The original status was already received above.
And after this code piece I don't see any additional exchange to confirm that handshake was successful